### PR TITLE
(ios) Sync encrypted contacts to iCloud as part of backup system

### PR DESCRIPTION
**Pull request by robbiehanson**
This PR updates the existing backup system (which was previously only used for payments) to also include contacts.

Very little is changed in terms of the UI. The section that was previously titled "Payments backup" is now called "Cloud backup". And the descriptions are changed to mention that both payments & contacts are backed up.

Just as we did when backing up payments, the contacts are also encrypted. That is, a "cloud key" is derived from the wallet's master key, and then the cloudKey is used with ChaCha20-Poly1305 to encrypt the serialized blob.

The contact photo is also backed up to the cloud. As mentioned in #600:
> we resize [contact photos] down to 1 megapixel, which results in a file size of around 100K.

This is nice as we will probably do more with the contact photo in the future.

### CODE REVIEW

**Review by dpad85 — COMMENTED** [review]
Looks good, I was able to test a reset successfully. By the way the checkbox in the Reset wallet screen mentions that only the payments history will be removed, I think it's a mistake.

**Comment by robbiehanson** [comment]
> By the way the checkbox in the Reset wallet screen mentions that only the payments history will be removed, I think it's a mistake.

Good catch. Fixed in 9411921

File: phoenix-ios/phoenix-ios/sync/SyncBackupManager+Contacts.swift [hunk 1]
```diff
@@ -0,0 +1,914 @@
...
+		)
+	}
+	
+	private func recordID(contactId: Lightning_kmpUUID) -> CKRecord.ID {
+		
+		// The recordID is:
+		// - deterministic => by hashing the contactId
+		// - secure => by mixing in the secret cloudKey (derived from seed)
+		
+		let prefix = SHA256.hash(data: cloudKey.rawRepresentation)
```
[the remark this answers is not in the document]
Comment (robbiehanson) [reply phoenix-ios/phoenix-ios/sync/SyncBackupManager+Contacts.swift]: It looks like for the database name we're doing this (in DatabaseManager.kt):
```kotlin
val nodeIdHash = nodeParams.nodeId.hash160().byteVector().toHex()
val channelsDb = SqliteChannelsDb(
  driver = createChannelsDbDriver(ctx, chain, nodeIdHash)
)
```

So I updated it to also use the nodeIdHash [ba71501](https://github.com/ACINQ/phoenix/pull/617/commits/ba715017df357526e679fd1f57e73f28a0064f4f)

**Comment by robbiehanson** [comment]
Note: I made some breaking changes in [e6091d4](https://github.com/ACINQ/phoenix/pull/617/commits/e6091d472840c6ebca0f2ec525ada8e624fddd2b). So if you have contacts currently stored in the cloud, then after pulling this change, you won't be able to pull them down on a restore.

What I did: I meant to store both payments & contacts in the same record zone (but different databases). But I was actually putting them in different record zones.

**Review by dpad85 — APPROVED** [review]

**Pull request merged**